programs/marinade-finance/src/instructions/user/withdraw_stake_account.rs

```diff
@@ -0,0 +1,303 @@
+        );
+        // require remainder stake also >= self.state.stake_system.min_stake
+        // To simplify the flow, we always deliver the lamports in the splitted account,
+        // so some lamports must remain in the original account. Check that
+        // after split, the amount remaining in the stake account is >= state.stake_system.min_stake
+        require_gte!(
+            stake
+                .last_update_delegated_lamports
+                .saturating_sub(split_lamports),
+            self.state.stake_system.min_stake,
+            MarinadeError::StakeAccountRemainderTooLow
+        );
+
+        // burn mSOL
+        burn(
+            CpiContext::new(
+                self.token_program.to_account_info(),
+                Burn {
+                    mint: self.msol_mint.to_account_info(),
+                    from: self.burn_msol_from.to_account_info(),
+                    authority: self.burn_msol_authority.to_account_info(),
+                },
+            ),
+            msol_amount,
+        )?;
+        self.state.on_msol_burn(msol_amount)?;
+
+        // split split_lamports from stake account into out split_stake_account
+        msg!(
+            "Split {} lamports from stake {} into {}",
+            split_lamports,
+            stake.stake_account,
+            self.split_stake_account.key(),
+        );
+
+        let split_instruction = stake::instruction::split(
+            self.stake_account.to_account_info().key,
+            self.stake_withdraw_authority.key,
+            split_lamports,
+            &self.split_stake_account.key(),
+        )
+        .last()
+        .unwrap()
+        .clone();
+        invoke_signed(
+            &split_instruction,
+            &[
+                self.stake_program.to_account_info(),
+                self.stake_account.to_account_info(),
+                self.split_stake_account.to_account_info(),
+                self.stake_withdraw_authority.to_account_info(),
+            ],
+            &[&[
+                &self.state.key().to_bytes(),
+                StakeSystem::STAKE_DEPOSIT_SEED,
+                &[self.state.stake_system.stake_deposit_bump_seed],
+            ]],
+        )?;
+
+        stake.last_update_delegated_lamports -= split_lamports;
+
+        // we now consider amount no longer "active" for this specific validator
+        validator.active_balance -= split_lamports;
+        // and in state totals,
+        self.state.validator_system.total_active_balance -= split_lamports;
+
+        // update stake-list & validator-list
+        self.state.stake_system.set(
+            &mut self.stake_list.data.as_ref().borrow_mut(),
+            stake_index,
+            stake,
+        )?;
+        self.state.validator_system.set(
+            &mut self.validator_list.data.as_ref().borrow_mut(),
+            validator_index,
+            validator,
+        )?;
+
+        // assign user as withdrawer (owner) & staker for the split_stake_account
```

Comment:
You must change staker auth first because if you set withdrawer first you are not an owner of the stake anymore and the second CPI will fail because it is expecting the user to be a signer instead of a withdrawer PDA